main.tf

```diff
@@ -7,35 +7,68 @@
 // - A database subnet group
 // - You should want your RDS instance in a VPC
 
-// Provider specific configs
-provider "aws" {
-    access_key = "${var.aws_access_key}"
-    secret_key = "${var.aws_secret_key}"
-    region = "${var.aws_region}"
-}
-
 resource "aws_db_instance" "main_rds_instance" {
-    identifier = "${var.rds_instance_name}"
+    identifier = "${var.rds_instance_identifier}"
     allocated_storage = "${var.rds_allocated_storage}"
     engine = "${var.rds_engine_type}"
     engine_version = "${var.rds_engine_version}"
     instance_class = "${var.rds_instance_class}"
     name = "${var.database_name}"
     username = "${var.database_user}"
     password = "${var.database_password}"
-    // Because we're assuming a VPC, we use this option, but only one SG id
-    vpc_security_group_ids = ["${var.rds_security_group_id}"]
-    // We're creating a subnet group in the module and passing in the name
+
+    # Because we're assuming a VPC, we use this option, but only one SG id
+    vpc_security_group_ids = ["${aws_security_group.main_db_access.id}"]
+
+    # We're creating a subnet group in the module and passing in the name
     db_subnet_group_name = "${aws_db_subnet_group.main_db_subnet_group.name}"
-    parameter_group_name = "${var.db_parameter_group}"
-    // We want the multi-az setting to be toggleable, but off by default
+    parameter_group_name = "${aws_db_parameter_group.main_rds_instance.id}"
+
+    # We want the multi-az setting to be toggleable, but off by default
     multi_az = "${var.rds_is_multi_az}"
     storage_type = "${var.rds_storage_type}"
     publicly_accessible = "${var.publicly_accessible}"
 }
 
+resource "aws_db_parameter_group" "main_rds_instance" {
+    name = "${var.rds_instance_identifier}-custom-params"
+    family = "${var.db_parameter_group}"
+
+    # Example for MySQL
+    # parameter {
+    #   name = "character_set_server"
+    #   value = "utf8"
+    # }
+
+    # parameter {
+    #   name = "character_set_client"
+    #   value = "utf8"
+    # }
+}
+
 resource "aws_db_subnet_group" "main_db_subnet_group" {
-    name = "${var.rds_instance_name}-subnetgrp"
+    name = "${var.rds_instance_identifier}-subnetgrp"
     description = "RDS subnet group"
-    subnet_ids = ["${var.subnet_az1}", "${var.subnet_az2}"]
+    subnet_ids = ["${var.subnets}"]
+}
+
+# Security groups
+resource "aws_security_group" "main_db_access" {
```

Comment:
Thanks! I'm on it.
Additionally I've spotted that db_parameter_group has set a default which might be not desired if we're not using assumed DB. I'll update it as well shortly.